### PR TITLE
Strings Format: Use context as key

### PR DESCRIPTION
**Pull request by akirk**
When importing from a .strings file we read the lines as follows (which is correct):

```
/* $comment */
"$context" = "$singular";
```

But when exporting, we export like this:
```
/* $comment */
"$singular" = "$translation";
```

when it should be 

```
/* $comment */
"$context" = "$translation";
```

This patch fixes this inconsistency.

### CODE REVIEW

File: gp-includes/formats/format-strings.php [hunk 1]
```diff
@@ -66,15 +66,15 @@ public function print_exported_file( $project, $locale, $translation_set, $entri
 		foreach ( $sorted_entries as $entry ) {
 			$translation = $this->escape( empty( $entry->translations ) ? $entry->singular : $entry->translations[0] );
 
-			$original    = str_replace( "\n", "\\n", $this->escape( $entry->singular ) );
+			$context    = str_replace( "\n", "\\n", $this->escape( $entry->context ) );
```
Comment (ocean90) [inline gp-includes/formats/format-strings.php]:
```suggestion
			$context     = str_replace( "\n", "\\n", $this->escape( $entry->context ) );
```

**Review by ocean90 — CHANGES_REQUESTED** [review]
Thanks @akirk! I guess this wasn't noticed because in [our test file](https://github.com/GlotPress/GlotPress-WP/blob/develop/tests/phpunit/data/originals.strings) the original and context is always the same? Can we add one which covers this issue?

**Comment by AliSoftware** [comment]
Hello @akirk @ocean90 👋  It seems that this PR had the unfortunate consequence of reintroducing #921 – which was fixed by #922 back then, and is where the use of the singular for the key instead of context was introduced.

As a matter of fact, I recently encountered the issue mentioned in #921 (and it took me a while to understand why I had it given that I found #922 that was supposed to have fixed it) for one of our keys that is longer than 255 characters.

**Comment by akirk** [comment]
Please take a look at [my response on that ticket](https://github.com/GlotPress/GlotPress-WP/issues/921#issuecomment-753828220). Short version is: there is no need to use the English text as a context, you can use anything shorter, you just need to specify an "English translation."

**Pull request merged**